packages/tempus/src/Tempus_StepperDIRK_impl.hpp

```diff
@@ -97,8 +97,10 @@ void StepperDIRK<Scalar>::setObserver(
   if (( obs == Teuchos::null ) and (this->stepperObserver_->getSize() == 0) )
      obs = Teuchos::rcp(new StepperRKObserver<Scalar>());
 
-  this->stepperObserver_->addObserver(
-       Teuchos::rcp_dynamic_cast<StepperRKObserver<Scalar> > (obs, true) );
+  // Check that this casts to prevent a runtime error if it doesn't
+  if (Teuchos::rcp_dynamic_cast<StepperRKObserver<Scalar> > (obs) != Teuchos::null)
+    this->stepperObserver_->addObserver(
+         Teuchos::rcp_dynamic_cast<StepperRKObserver<Scalar> > (obs, true) );
```

Comment:
This is dangerous in that the check allows you to silently ignore registered observers. I can see people trying to debug asking "why when I call addObserver does it silently not add it?"

Reply:
I can add a warning to that effect; the code right now dies at runtime if you don't do this. Alternatively, we can remove it after the downstream app has done the promotion.

Reply:
I would probably lean towards leaving this as is to get the app up and running in the short term. @ccober6 any preference? We should set up a meeting to discuss a long term design of observers. They seem to be the source of many complications.

Reply:
Yeah, I agree.  My branch that is on hold simplifies this setObserver(), and the primary reason I wanted to get my branch through.  FYI, the changes that @sconde put into EMPIRE-Hybrid will need to be reverted with my branch.  This is an unfortunate situation where Sidafa changes were not the path we wanted and the next commit changes it to something better, but currently we are in-between those changes.

Reply:
Ok, so we're all agreed that we put this in and then improve it later?

Reply:
I would probably lean towards adding print statement noting that an observer was not added because of casting problem.  But otherwise yes let's make the change and improve later.

Reply:
Warnings have been added.